src/Router.tsx

```diff
@@ -47,9 +45,27 @@ export function Router() {
           </GuardedRoute>
         }
       />
-      {/* Questionnaire route guarded by authentication, with optional questionId parameter */}
+      {/* Use exact matching for results */}
       <Route
-        path={`${getFullPath(RoutePaths.QUESTIONNAIRE)}/:questionId?`}
+        path={getFullPath(RoutePaths.RESULTS)}
+        element={
+          <GuardedRoute>
+            <Results />
+          </GuardedRoute>
+        }
+      />
+      {/* Use a more specific match for the questionnaire with optional ID */}
+      <Route
+        path={`${getFullPath(RoutePaths.QUESTIONNAIRE)}/:questionId`}
```

Comment:
You shouldn't have to define two routes for the same component. 
There is no `/questionnaire` page actually. When you click on the Questionnaire from the top bar, it takes me to the first question. Then instead of showing the `/questionnaire` path, why not directly take the user to the `/questionnaire/1` path?